integration_tests/tests/billing-and-transaction-view-tests.ts

```diff
@@ -84,22 +77,79 @@ interface TestData {
   eventName: EventName;
   vendorId: VendorId;
   eventTime: TimeStamps;
+  unitPrice: number;
   numberOfTestEvents: number;
   priceDiff: string;
   qtyDiff: string;
   priceDifferencePercent: string;
   qtyDifferencePercent: string;
   billingPrice: string;
-  billingQty: string;
+  billingQty: number;
   transactionPrice: string;
   transactionQty: string;
 }
 
-export const assertResultsWithTestData = async ({
+export const createInvoice = async ({
   eventName,
   vendorId,
   eventTime,
+  unitPrice,
+  billingQty,
+}: TestData): Promise<void> => {
+  const givenVendorId = vendorId;
+  const givenEventName = eventName;
+  const givenBillingQty = billingQty;
+
+  const lineItems = randomLineItem({
+    description: "passport check",
+    quantity: givenBillingQty,
+    unitPrice,
+  });
+
+  const vendorServiceConfigRow = await getVendorServiceConfigRow(
+    configStackName(),
+    { vendor_id: givenVendorId, event_name: givenEventName }
+  );
+
+  const givenInvoice = randomInvoice({
+    vendor: {
+      name: vendorServiceConfigRow.vendor_name,
+    },
+    date: new Date(eventTimeStamp[eventTime] * 1000),
+    lineItems: [lineItems],
+  });
+
+  const filename = `raw-Invoice-${Math.random()
+    .toString(36)
+    .substring(2, 7)}-validFile.pdf`;
+  console.log("raw-invoice-filename:", filename);
+  await createInvoiceInS3({ invoiceData: givenInvoice, filename });
+
+  const res = await poll(
+    async () =>
+      await listS3Objects({
+        bucketName,
+        prefix: "btm_billing_standardised",
+      }),
+    ({ Contents }) =>
+      !!Contents?.some(
+        (s3Object) =>
+          s3Object.LastModified !== undefined &&
+          new Date(s3Object.LastModified) >= testStartTime
+      ),
+    {
+      timeout: 50000,
+      nonCompleteErrorMessage:
+        "Invoice data never appeared in standardised folder",
+    }
+  );
+  console.log("**********", res);
```

Comment:
I think perhaps these three console.logs can be removed now.

Reply:
done